text/0061-error-format-and-definitions.md

```diff
@@ -0,0 +1,807 @@
+
+### Error Definition
+
+HTTP Code: `500 Internal Server Error``
+
+```json
+{
+    "message": "An internal error has occurred. :reason.",
+    "code": "internal",
+    "type": "internal",
+    "link": "https://docs.meilisearch.com/errors#internal"
+}
+```
+
+- The `:reason` is inferred when the message is generated. e.g. `Internal: decoding failed`
+
+---
+
+## index_creation_failed
+
+### Context
+
+This error occurs when an index creation could not be completed for various reasons.
+
+### Error Definition
+
+HTTP Code: `500 Internal Server Error`
+
+```json
+{
+    "message": "The creation of the :uid index has failed due to :reason",
+    "code": "index_creation_failed",
+    "type": "internal",
+    "link": "https://docs.meilisearch.com/errors#index_creation_failed"
+}
+```
+
+- The `:reason` is inferred when the message is generated. e.g. `Write Permission denied`
+
+---
+
+## index_not_accessible
+
+### Context
+
+This error occurs when an index can be accessed for various reasons.
+
+### Error Definition
+
+HTTP Code: `500 Internal Server Error`
+
+```json
+{
+    "message": "The index :uid can't be accessed due to :reason",
+    "code": "index_not_accessible",
+    "type": "internal",
+    "link": "https://docs.meilisearch.com/errors#index_not_accessible"
+}
+```
+
+- The `:reason` is inferred when the message is generated. e.g. `Read Permission denied`
+
+---
+
+## unretrievable_document
+
+### Context
+
+This error occurs when a document cannot be found in the system due to an inconsistent state that can occur for several reasons.
+
+### Error Definition
+
+HTTP Code: `500 Internal Server Error`
+
+```json
+{
+    "message": "The document :documentId is unretrievable due to :reason.",
+    "code": "unretrievable_document",
+    "type": "internal",
+    "link": "https://docs.meilisearch.com/errors#unretrievable_document"
+}
+```
+
+- The `documentId` is inferred when the message is generated.
+- The `:reason` is inferred when the message is generated.
+
+---
+
+## invalid_state
+
+### Context
+
+This error occurs when the database is in an inconsistent state due to an uncontrolled internal error.
+
+### Error Definition
+
+HTTP Code: `500 Internal Server Error`
+
+```json
+{
+    "message": "The database is in an invalid state.",
+    "code": "invalid_state",
+    "type": "internal",
+    "link": "https://docs.meilisearch.com/errors#invalid_state"
+}
+```
+
+---
+
+## dump_process_failed
+
+### Context
+
+This error occurs during the dump creation process. The dump creation was interrupted for various reasons.
+
+💡 Errors returned asynchronously in a `dump` object, does not include a definition of the http code.
+
+### Error Definition
+
+```json
+{
+    "message": "The creation of the dump :dumpId failed due to :reason.",
+    "code": "dump_process_failed",
+    "type": "internal",
+    "link": "https://docs.meilisearch.com/errors#dump_process_failed"
+}
+```
+
+- The `:reason` is inferred when the message is generated.
```

Comment:
What is an example of a possible `:reason`?

Reply:
For example, it can be a denied permission for writing to the file system. I'll add it to give more context. In general, known internal errors can occur as a result of system misconfiguration or limits being reached.